### PR TITLE
Add better logging for building errors

### PR DESCRIPTION
**Pull request by someaddons**
# Changes proposed in this pull request:
-Add better logging for building errors

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 1]
```diff
@@ -197,7 +197,7 @@ public <T extends IBuildingModule> T getFirstModuleOccurance(final Class<T> claz
             }
         }
 
-        throw new IllegalStateException("The module of class: " + clazz.toString() + "should never be null!");
+        throw new IllegalStateException("The module of class: " + clazz.toString() + "should never be null! Building:"+getBuildingType().getTranslationKey()+" pos:"+getID());
```
Comment (uecasm) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: It's a little nicer to use `.toShortString()` when logging `BlockPos`.

File: src/main/java/com/minecolonies/coremod/colony/buildings/modules/WorkAtHomeBuildingModule.java [hunk 2]
```diff
@@ -46,8 +46,8 @@ public boolean assignCitizen(final ICitizenData citizen)
                         buildingComponent,
                         BlockPosUtil.getString(oldHome.getID()))
                       .sendTo(oldHome.getColony()).forAllPlayers();
+                    oldHome.getFirstModuleOccurance(LivingBuildingModule.class).removeCitizen(citizen);
                 }
-                oldHome.getFirstModuleOccurance(LivingBuildingModule.class).removeCitizen(citizen);
```
Comment (uecasm) [inline src/main/java/com/minecolonies/coremod/colony/buildings/modules/WorkAtHomeBuildingModule.java:50]: This seems suspicious.  Won't this leave the citizen assigned to two modules if the `if` condition fails (e.g. the old home does have a `WorkAtHomeBuildingModule`)?

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/buildings/modules/WorkAtHomeBuildingModule.java:50]: if (oldHome.hasModule(LivingBuildingModule.class) && !oldHome.hasModule(WorkAtHomeBuildingModule.class))

I think the second part of this if condition shouldn't be. Else assigning from one guard tower to another can cause issues.

Comment (MotionlessTrain) [reply src/main/java/com/minecolonies/coremod/colony/buildings/modules/WorkAtHomeBuildingModule.java:50]: That part of the `if` condition looks to be fine for the chat message (saying that a house has a free bed now), possible not for the rest of the block statement

Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/buildings/modules/WorkAtHomeBuildingModule.java:50]: yeah, should have the oldHome has module for the remove check, and the second one for the print message

**Pull request merged**